# **Chi-Square test**

### **Objective:**
To use the Chi-Square test for independence to determine if there's a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.

### **1. State the Hypotheses:**



*   **Null Hypothesis (H₀)**: Customer satisfaction is independent of the device type.
*   **Alternative Hypothesis (H₁)**: Customer satisfaction is dependent on the device type.



### **2. Compute the Chi-Square Statistic:**

In [ ]:
#import required libraries
import numpy as np
from scipy.stats import chi2_contingency

In [ ]:
# Observed frequencies
a=np.array([[50,70],[80,100],[60,90],[30,50],[20,50]])

In [ ]:
chi2,p,dof,excepted=chi2_contingency(a)
print("chi2=",chi2)
print("p=",p)
print("dof=",dof)
print("excepted=",excepted)

chi2= 5.638227513227513
p= 0.22784371130697179
dof= 4
excepted= [[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


### **3. Determine the Critical Value:**

The critical value is determined by the degrees of freedom and the significance level (alpha = 0.05). The degrees of freedom (dof) can be calculated as:

dof
=
(
Number of Rows
−
1
)
×
(
Number of Columns
−
1)


In [ ]:
degrees_of_freedom=(a.shape[0]-1)*(a.shape[1]-1)
degrees_of_freedom

4

In [ ]:
import scipy.stats
# Significance level
alpha = 0.05

# Critical value from Chi-Square distribution with the calculated degrees of freedom
critical_value = scipy.stats.chi2.ppf(1 - alpha, 4)
print("Critical value:", critical_value)

Critical value: 9.487729036781154


### **4. Make a Decision:**



Compare the Chi-Square statistic with the critical value to decide whether to accept the null hypothesis or reject the null hypothesis

In [ ]:
if chi2>=critical_value:
    print("reject the null hypothesis")
else:
    print("accept the null hypothesis")

accept the null hypothesis


### **5.Conclusion:**

Since the Chi-Square statistic (5.64) is less than the critical value (9.49) and the p-value (0.23) is greater than the significance level (0.05), we fail to reject the null hypothesis. This suggests that there is no significant association between the type of smart home device and customer satisfaction levels based on the data provided

# **Hypothesis Testing**

### **Objective:**
To investigate the restaurant owners' claim about the increase in weekly operating costs using hypothesis testing.

###**1. State the Hypotheses statement:**###



*   **Null Hypothesis (H₀):** The mean weekly operating cost according to the theoretical model is accurate.  
H₀
:
𝜇
=
1000+5X

*   **Alternative Hypothesis (H₁):** The mean weekly operating cost is higher than the theoretical model suggests.  
H₁
:
𝜇 >
1000+5X




### **2. Calculate the Test Statistic:**

Given   
xˉ = sample mean weekly cost (Rs. 3,050)  
μ = theoretical mean weekly cost according to the cost model (W = $1,000 + $5X for X = 600 units)  
σ = 5*25 units  
n = sample size (25 restaurants)

In [ ]:
import scipy.stats as stats

# Given data
sample_mean = 3050  # Sample mean weekly cost (Rs.)
theoretical_mean = 1000 + 5 * 600  # Theoretical mean weekly cost
sigma = 5 * 25  # Standard deviation of weekly cost
n = 25  # Sample size

# Calculate the test statistic
t_statistic = (sample_mean - theoretical_mean) / (sigma / n**0.5)
print("Test Statistic:", t_statistic)

Test Statistic: -38.0


### **3. Determine the Critical Value:**
Using the alpha level of 5% (α = 0.05), determine the critical value from the standard normal (Z) distribution table

In [ ]:
# Significance level
alpha = 0.05

# Degrees of freedom
dof = n - 1

# Critical value from the t-distribution table for one-tailed test
t_critical = stats.t.ppf(1 - alpha, dof)
print("Critical Value:", t_critical)

Critical Value: 1.7108820799094275


### **4. Make a Decision:**
Compare the test statistic with the critical value to decide whether to reject the null hypothesis or accept the null hypothesis

In [ ]:
if t_statistic > t_critical:
    print("reject the null hypothesis")
else:
    print("accept the null hypothesis")

accept the null hypothesis


### **5. Conclusion:**

Since the test statistic (-38.0) is much less than the critical value (1.71), we fail to reject the null hypothesis.
There is no strong evidence to support the restaurant owners' claim that the weekly operating costs are higher than the model suggests based on the provided data.